src/ServiceAccountCredentials.php

```diff
@@ -104,4 +104,102 @@ public function getCacheKey()
     }
     return $key;
   }
+
+  /**
+   * Updates metadata with the authorization token
+   *
+   * @param $metadata array metadata hashmap
+   * @param $authUri string optional auth uri
+   * @param $client optional client interface
+   *
+   * @return array updated metadata hashmap
+   */
+  public function updateMetadata($metadata,
+                                 $authUri = null,
+                                 ClientInterface $client = null)
+  {
+    // scope exists. use oauth implementation
+    $scope = $this->auth->getScope();
+    if (!is_null($scope)) {
+      return parent::updateMetadata($metadata, $authUri, $client);
+    }
+
+    // no scope found. create jwt with the auth uri
+    $credJson = array(
+      'private_key' => $this->auth->getSigningKey(),
+      'client_email' => $this->auth->getIssuer(),
+    );
+    $jwtCreds = new ServiceAccountJwtAccessCredentials($credJson);
+    return $jwtCreds->updateMetadata($metadata, $authUri, $client);
+  }
+}
+
+/**
+ * Authenticates requests using Google's Service Account credentials via
+ * JWT Access.
+ *
+ * This class allows authorizing requests for service accounts directly
+ * from credentials from a json key file downloaded from the developer
+ * console (via 'Generate new Json Key').  It is not part of any OAuth2
+ * flow, rather it creates a JWT and sends that as a credential.
+ */
+class ServiceAccountJwtAccessCredentials extends CredentialsLoader
```

Comment:
I would add this class in a different file. At the same time now that there are two different ServiceAccountCredentials an interface could make sense with the method  updateMetadata